src/main/java/com/autotune/experimentManager/core/ExperimentTrialHandler.java

```diff
@@ -153,27 +157,69 @@ public void startExperimentTrials() {
+                                // for now just like deployment handler we run a for loop and constantly poll for the load availability
+                                for (int j = 0; j < EMConstants.StandardDefaults.BackOffThresholds.CHECK_LOAD_AVAILABILITY_THRESHOLD; j++) {
+                                    // Proceed to check if load is available (minimal variation)
+                                    if (EMUtil.InterceptorAvailabilityStatus.AVAILABLE == currentAvailability) {
+                                        // Will proceed for metric cycles if the load is detected
+                                        proceedToMetricsCollection = true;
+                                        // breaking the load availability loop
+                                        break;
+                                    }
+                                    try {
+                                        LOGGER.debug("The Load is not yet available, will be checking it again");
+                                        // Will be replaced by a exponential looper mechanism
+                                        Thread.sleep(EMUtil.timeToSleep(j) * 1000);
+                                    } catch (InterruptedException e) {
+                                        e.printStackTrace();
+                                    }
+                                    currentAvailability = emLoadInterceptor.isAvailable(this.experimentTrial);
+                                }
+                                if (EMUtil.InterceptorAvailabilityStatus.AVAILABLE != currentAvailability) {
+                                    LOGGER.debug("Load cannot be detected for the particular trial, proceed to collect metrics if it can be ignored");
+                                    if (this.experimentTrial.getExperimentSettings().getTrialSettings().isForceCollectMetrics()) {
+                                        proceedToMetricsCollection = true;
+                                    }
+                                }
+                            } else {
+                                LOGGER.debug("Load cannot be detected for the particular trial, proceed to collect metrics if it can be ignored");
+                                if (this.experimentTrial.getExperimentSettings().getTrialSettings().isForceCollectMetrics()) {
+                                    proceedToMetricsCollection = true;
+                                }
+                            }
+                            if (!proceedToMetricsCollection) {
+                                LOGGER.debug("Proceeding to next trial and this trial cannot be completed due to lack of metrics collection. Will be marked as failed trial");
+                                // Need to implement a exiting functionality
+                            } else {
+                                //collect warmup and measurement cycles metrics
+                            }
```

Comment:
Why is this code here and not in the load abstraction? It should just be a simple call here to check whether we move forward or error out